security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -220,23 +220,20 @@ func (s *sdsservice) StreamSecrets(stream sds.SecretDiscoveryService_StreamSecre
 				return receiveError
 			}
 
-			if discReq.Node != nil {
+			if discReq.Node == nil {
```

Comment:
I doubt this is intended behavior...

Reply:
I mean: when and why would it be nil, and if it is - why should it override a previous not-nil value ?

Can you add comments to explain it ?

Reply:
Node in xDS request is delta compressed. First one is set, subsequent are omitted. See the note from the spec in the description.